src/model/entity/level.go

```diff
@@ -1,8 +1,98 @@
 package entity
 
+import (
+	"errors"
+	"math/rand"
+)
+
+const (
+	RoomMinWidth     = 3
+	RoomMinHeight    = 3
+	MinRoomPadding   = 1
+	numberXYSections = 3
+)
+
 type Level struct {
-	Rooms       []Room
-	Passages    []Passage
-	LevelNumber uint
-	LevelEnd    Box
+	Rooms    []Room
+	Passages []Passage
+	Number   uint
+	End      Box
+}
+
+func calculateRoomSection(mapSize Size2D[uint]) (sectionSize Size2D[uint]) {
```

Comment:
Вот тут из названия `size` лучше было не убирать, даже если в аргументах дублируется.


```suggestion
func calculateRoomSectionSize(mapSize Size2D[uint]) (sectionSize Size2D[uint]) {
```